models/note.go

```diff
@@ -2,51 +2,81 @@ package models
 
 import (
 	"encoding/json"
+	"errors"
+	"strings"
 	"time"
 )
 
 type NoteId int64
 
 type NoteType int
 
+var InvalidNoteTypeError = errors.New("Note type does not exist")
+
+var UnDeserializeableNoteTypeStringError = errors.New("String does not correspond to a NoteType")
+
 const (
-	MARGINALIA NoteType = iota
+	UNCATEGORIZED NoteType = iota
+	MARGINALIA
 	META
 	QUESTIONS
 	PREDICTIONS
 )
 
 var noteTypeStrings = [...]string{
-	"Marginalia",
-	"Meta",
-	"Questions",
-	"Predictions",
+	"uncategorized",
+	"marginalia",
+	"meta",
+	"questions",
+	"predictions",
+}
+
+func DeserializeNoteType(input string) (NoteType, error) {
+	for i := 0; i < len(noteTypeStrings); i++ {
+		if strings.ToLower(input) == strings.ToLower(noteTypeStrings[i]) {
+			return NoteType(i), nil
+		}
+	}
+	return UNCATEGORIZED, UnDeserializeableNoteTypeStringError
```

Comment:
I hate putting a value here like uncategorized, however, the go compiler wouldn't allow me to put nil.